### DISCUSSION TITLE
Puma refuses to compile.

### DISCUSSION THREAD

**Discussion by xinghengwang** [puma/puma#3070] [Q&A]
**Describe the bug**

```
gem install puma
Building native extensions. This could take a while...
ERROR:  Error installing puma:
	ERROR: Failed to build gem native extension.

    current directory: /.rvm/gems/ruby-3.1.2/gems/puma-6.0.2/ext/puma_http11
/.rvm/rubies/ruby-3.1.2/bin/ruby -I /Users/Xingheng/.rvm/rubies/ruby-3.1.2/lib/ruby/3.1.0 -r ./siteconf20230201-86752-4xrvkf.rb extconf.rb
using OpenSSL pkgconfig (openssl.pc)
checking for openssl/bio.h... yes
checking for DTLS_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_session_cache_mode(NULL, 0) in openssl/ssl.h... yes
checking for TLS_server_method() in openssl/ssl.h... yes
checking for SSL_CTX_set_min_proto_version(NULL, 0) in openssl/ssl.h... yes
checking for X509_STORE_up_ref()... yes
checking for SSL_CTX_set_ecdh_auto(NULL, 0) in openssl/ssl.h... yes
checking for SSL_CTX_set_dh_auto(NULL, 0) in openssl/ssl.h... yes
checking for SSL_get1_peer_certificate() in openssl/ssl.h... no
checking for Random.bytes... yes
creating Makefile
````

**Puma config:**

or `rails start` always stuck on puma install. 

so I tried using `gem install puma` 
Still fails as above. 

Please copy-paste your Puma config AND your command line options here.

MacOS 13.1
Xcode 14. 
rvm installed ruby 3.1.2

no configuraiton. 
just simple 
`gem install puma`

**To Reproduce**

See above. 

```
run lambda { |env| [200, {"Content-Type" => "text/plain"}, ["Hello World"]] }
```

Run it with:

```
bundle exec puma -C <where_your_config_is> hello.ru
```

**Desktop (please complete the following information):**
 - OS: MacOS 13.1
 - Puma Version latest or 5.6 as rails 7 needed.

**Comment by MSP-Greg** [comment]
In CI, we can only easily test with MacOS 12.  But, your log shows:
```
checking for SSL_get1_peer_certificate() in openssl/ssl.h... no
```
The CI shows:
```
checking for SSL_get1_peer_certificate() in openssl/ssl.h... yes
```

Seems to indicate that your system has OpenSSL 1.1.1, but our CI systems have OpenSSL 3.0.x.  I don't run MacOS locally, so I can't help much.  Some people have both OpenSSL 1.1.1 & 3.0.x installed, which can cause problems.

I recall something about openssl.pc causing issues with two installs, but can't recall how.

Maybe either:
```
gem install puma -- --with-openssl-dir=<not sure>
or
bundle config build.puma --with-openssl-dir=<not sure>
```

**Comment by MSP-Greg** [comment]
Also, to verify that it is not an OpenSSL problem, you can set PUMA_DISABLE_SSL, and it should compile without SSL/TLS support.
